File: 2016-05-08-Android-Overlay-Permission-Example/src/main/java/tech/thdev/android_overlay_permission_example/base/presenter/BasePresenter.java
package tech.thdev.android_overlay_permission_example.base.presenter;

import tech.thdev.android_overlay_permission_example.base.presenter.view.BaseView;

/**
 * Created by tae-hwan on 5/8/16.
 */
public abstract class BasePresenter<T extends BaseView> {

    private T view;

    public BasePresenter(T view) {
        this.view = view;
    }

    protected T getView() {
        return view;
    }
}

File: kotlin-example-01/src/main/kotlin/tech/thdev/kotlin_example_01/base/presenter/BasePresenter.kt
package tech.thdev.kotlin_example_01.base.presenter

/**
 * Created by Tae-hwan on 7/21/16.
 */
interface BasePresenter<VIEW: BaseView<*>> {

    /**
     * View Attach.
     */
    fun attachView(view: VIEW)

    /**
     * View detach
     */
    fun detachView()

    /**
     * GetView
     */
    fun getView(): VIEW?
}